### PR TITLE
Bluetooth: Audio: Fix issues the unicast client creating groups before configuring streams

### PR DESCRIPTION
**Pull request by Thalley**
This PR fixes an issue with releasing streams, as well as an issue that prohibited the unicast client from creating the unicast group before configuring the streams. 

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

### CODE REVIEW

**Review by koffes — DISMISSED** [review]

File: subsys/bluetooth/audio/audio_iso.c [hunk 1]
```diff
@@ -242,3 +234,72 @@ struct bt_audio_ep *bt_audio_iso_get_ep(bool unicast_client,
 		return iso->tx.ep;
 	}
 }
+
+#if defined(CONFIG_BT_AUDIO_UNICAST_CLIENT)
+void bt_audio_iso_bind_stream(struct bt_audio_iso *audio_iso,
+			      struct bt_audio_stream *stream)
+{
+	struct bt_audio_iso_dir *audio_iso_ep;
```
Comment (MariuszSkamra) [inline subsys/bluetooth/audio/audio_iso.c:252]: s/audio_iso_ep/audio_iso_dir

File: subsys/bluetooth/audio/audio_iso.c [hunk 2]
```diff
@@ -242,3 +234,72 @@ struct bt_audio_ep *bt_audio_iso_get_ep(bool unicast_client,
...
+		 audio_iso, audio_iso_ep->stream);
+	audio_iso_ep->stream = stream;
+
+	stream->audio_iso = bt_audio_iso_ref(audio_iso);
+}
+
+void bt_audio_iso_unbind_stream(struct bt_audio_iso *audio_iso,
+				struct bt_audio_stream *stream)
+{
+	struct bt_audio_iso_dir *audio_iso_ep;
```
Comment (MariuszSkamra) [inline subsys/bluetooth/audio/audio_iso.c:281]: ditto

**Review by MariuszSkamra — DISMISSED** [review]
LGTM

**Comment by Thalley** [comment]
Fixed bad check for `if (ep->iso == NULL) {` which should have been (and now is) `if (stream->audio_iso == NULL) {` in `bt_audio_stream_qos`

**Comment by Thalley** [comment]
Added 2 commits to fix a build warning and revert a bad workaround to handle missing PHY values.

**Review by MariuszSkamra — APPROVED** [review]

**Comment by MariuszSkamra** [comment]
@sjanc Could you take a look at this one?

**Review by sjanc — APPROVED** [review]

**Pull request merged**